04-materials/02-anatomy-of-extensions.md

```diff
@@ -158,105 +123,254 @@ normally would!
+   ```
+   :::
+
+2. Set up the Git CLI to authenticate with GitHub:
 
    ```bash
    gh auth setup-git
    ```
 
-2. Create a repository in GitHub and clone it:
+3. Create a repository in GitHub and clone it:
 
    ```bash
    gh repo create jupytercon2025-extension-workshop --public --clone
    ```
 
-3. Change directory into your newly-cloned repository:
+4. Change directory into your newly-cloned repository:
 
    ```bash
    cd jupytercon2025-extension-workshop
    ```
 
-4. Add some useful metadata to your repository:
+5. Add some useful metadata to your repository:
 
    ```bash
    gh repo edit --add-topic "jupytercon2025" --add-topic "jupyterlab-extension"
    ```
 
-5. Get your cloned repository's URL:
+6. Get your cloned repository's URL:
 
    ```bash
    gh repo view
    ```
 
-   The URL of your repo will be printed on the final line of output.
-   Copy that URL for the next step!
+   :::{important} 👀 You should notice...
+   :class: simple
+   :icon: false
 
-:::{hint}
-Check out [GitHub authentication](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
-about SSH keys later if you get tired of typing username/password.
+   ...a full description of your repository is printed to the terminal.
+   The final line of output includes a repository URL starting with
+   `https://github.com/`.
+   :::
+
+   **Copy the entire repository URL for the next step!**
+
+
+## Extensions and plugins and widgets -- oh, my!
+
+While they sound similar, `extensions <extension>` and {term}`plugins <plugin>` serve
+different purposes.
+
+{term}`Plugins <plugin>` are JupyterLab's fundamental building blocks which define
+functionality and business logic.
+{term}`Extensions <extension>` are the delivery mechanism or "container" for plugins.
+Extensions are the thing that end-users `pip install`.
+
+:::{pull-quote}
+End-users care about extensions, and developers care about plugins.
 :::
 
+A {term}`widget <widget>` is a user interface component provided by a plugin, either for
+the end user to display (e.g. an interactive visualization of data) or for JupyterLab to
+display (e.g. a document viewer that opens when you double-click a particular file
+type).
+
+```{mermaid}
+graph TB
+
+    subgraph Extension["Extension"]
+        subgraph Plugin["Plugin(s) (n>=1)"]
+            Widget["Widget(s) (n>=0)"]
+        end
+    end
+
+    style Widget stroke-dasharray: 5 5
+```
+
+
+## What are we building together?
+
+First, we'll examine and demonstrate
+[the extension we're going to build today](https://github.com/jupytercon/jupytercon2025-developingextensions-demo).
+
+Our extension will:
+
+* Add a viewer for a small collection of public domain images and captions
+* Enable the user to interactively display a random item from the collection
+* Add a new button to the launcher to open the viewer
+* Add a new command to the command palette to open the viewer
+
```

Comment:
```suggestion
- Add a new button to refresh the image
```

Reply:
I covered that in this entry:

>* Enable the user to interactively display a random item from the collection

Would you suggest changing the wording?

Reply:
🤦‍♂️I can't read... You are good here